tools/ts_source_copy_check.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import argparse
+import logging
+import os
+import re
+import sys
+import typing
+
+from lxml import etree
+import githelper
+
+# Define the path to the allow list XML file
+ALLOW_LIST_PATH = "res/translations/source_copy_allow_list.tsv"
+PROPOSED_ALLOW_LIST_PATH = (
+    "res/translations/source_copy_allow_list_proposed.tsv"
+)
+
+
+def is_untranstaled_allowed(source_text, language):
+    if not os.path.exists(ALLOW_LIST_PATH):
+        return False
+
+    with open(ALLOW_LIST_PATH) as f:
+        for i, line in enumerate(f):
+            if not line.endswith("\n"):
+                print(f"Parse Error: No \\n at {ALLOW_LIST_PATH}:{i + 1}")
+                return False
+            line = line[:-1]
+            try:
+                lang, source = line.split("\t", 1)
+            except Exception as e:
+                print(f"Parse Error at {ALLOW_LIST_PATH}:{i + 1}")
+                print(e)
+            if lang is None or lang == "":
+                print(
+                    f"Parse Error: lang is empty at{ALLOW_LIST_PATH}:{i + 1}"
+                )
+                continue
+            source = source.encode("utf-8").decode("unicode_escape")
+            if source == source_text:
+                if lang == "*":
+                    return True
+                if language in lang:
```

Comment:
This means that `en` will also match if the string is `de,en_US,fr` although `en` its not in the list.

```suggestion
                if language in lang.split(","):
```

Even better would be to use `fnmatch` as I suggested before, because if you really want to match all English dialects, you could write `de,en*,fr` without having to list each and every one of them.